packages/api/endpoints/granules.js

```diff
@@ -140,37 +142,48 @@ async function del(req, res) {
   log.info(`granules.del ${granuleId}`);
 
   const granuleModelClient = new models.Granule();
+  const granulePgModel = new GranulePgModel();
+  const filePgModel = new FilePgModel();
 
-  let granule;
+  const knex = await getKnexClient({ env: process.env });
+
+  let dynamoGranule;
+  let pgGranule;
   try {
-    granule = await granuleModelClient.getRecord({ granuleId });
+    dynamoGranule = await granuleModelClient.getRecord({ granuleId });
+    pgGranule = await granulePgModel.get(knex, { granule_id: granuleId });
   } catch (error) {
     if (error instanceof RecordDoesNotExist) {
       return res.boom.notFound(error);
     }
     throw error;
   }
 
-  if (granule.detail) {
-    return res.boom.badRequest(granule);
+  // DEPRECATED this can be removed
+  if (dynamoGranule.detail) {
+    return res.boom.badRequest(dynamoGranule);
   }
 
-  try {
-    await granuleModelClient.delete(granule);
-  } catch (error) {
-    if (error instanceof DeletePublishedGranule) {
-      return res.boom.badRequest(error.message);
-    }
-    throw error;
+  if (pgGranule.published) {
```

Comment:
I had to pause and think about this for a moment.... it seems ok though, if a granule is *in* postgres, you can check the publish flag. 

However, given the comment on L154, this should be evaluated after that's resolved.

Reply:
Yup. This is switched to look at the Dynamo record's `published` status. I think that's what we would want if the PG granule is optional here.